packages/yarnpkg-core/sources/semverUtils.ts

```diff
@@ -82,3 +82,20 @@ export function validRange(potentialRange: string): semver.Range | null {
   rangesCache.set(potentialRange, range);
   return range;
 }
+
+/**
+ The RegExp from https://semver.org/ but modified to
+ - allow the version to start with `(?:[\Wv=]*?)`
+ - allow the version to end with `(?:\W*)`
+ - place the valid version in capture group one
+ */
+const CLEAN_SEMVER_REGEXP = /^(?:[\Wv=]*?)((0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)(?:\W*)$/;
```

Comment:
What's the reason for allowing versions such as `1.2.3!` (and cleaning them into `1.2.3`)? :thinking:

Reply:
Wow, don't know how/why I mixed up the tokens but it was supposed to be `\s`.